src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java

```diff
@@ -164,4 +171,25 @@ protected String getComid(String featureSource, String featureID) {
 			return feature.get(Parameters.COMID).toString();
 		}
 	}
+
+	protected String getHtml(String url) throws IOException {
+		String html = new String(FileCopyUtils.copyToByteArray(new ClassPathResource("static/controllers/json_link_template.html").getInputStream()));
+		return html.replace(URL_MARKER, url + "?f=json");
+	}
+
+	protected String resolveFormat(String format, String acceptHeader) {
```

Comment:
We should do this the Spring way - In these cases, I think using the @GetMapping's produces on two different methods (one for json, one for html). I REALLY do not think it is appropriate to produce HTML from the IOW data services. The navigation services are NOT designed for a paging interface. Not sure how they could be adapted in a reasonable generic way. The simple up/down stream main might be pagable (if a single flowline), but not the tributary navigations.

Reply:
Okay, I will fix the code.

Regarding the paging, as of now paging is not being discussed.  There is a github issue to create a query where the user provides latitude and longitude, and Dave Blodgett said maybe in that case we can return a leaflet and the user can see the points in the basin where that latitude and longitude is contained.  What to do for the navigation stuff is still very TBD.   But I will keep your words in mind.

Reply:
Code is fixed.